pages/deployment/[uuid].tsx

```diff
@@ -128,18 +134,18 @@ export default function Deployment() {
   if (error) return error
 
   if (isDeploymentLoading || isUiConfigLoading || isCurrentUserLoading) {
-    return <Wrapper title={'Loading...'} page={'deployment'} />
+    return <Wrapper title='Loading...' page='deployment' />
   }
 
-  const deploymentTag = `${uiConfig?.registry.host}/${currentUser!.id}/${tag}`
+  const deploymentTag = `${uiConfig?.registry.host}/${currentUser?.id}/${tag}`
 
   const requestApprovalReset = async () => {
     await postEndpoint(`/api/v1/deployment/${deployment?.uuid}/reset-approvals`, {}).then((res) => res.json())
   }
 
   return (
     <>
-      <Wrapper title={`Deployment: ${deployment!.metadata.highLevelDetails.name}`} page={'deployment'}>
+      <Wrapper title={`Deployment: ${deployment?.metadata.highLevelDetails.name}`} page='deployment'>
```

Comment:
If deployment is undefined, `highLevelDetails` on `deployment.metadata` will also be undefined, so this will break.  We should inform TypeScript that this is defined in other ways.

Reply:
This can be done using a Type Guard.